src/compiler/checker.ts

```diff
@@ -30341,7 +30353,12 @@ namespace ts {
                         checkTypeArgumentConstraints(node, typeParameters);
                     }
                 }
-                if (type.flags & TypeFlags.Enum && getNodeLinks(node).resolvedSymbol!.flags & SymbolFlags.EnumMember) {
+                const symbol = getNodeLinks(node).resolvedSymbol!;
+                if (symbol?.isDeprecated) {
+                    const diagLocation = isTypeReferenceNode(node) && isQualifiedName(node.typeName) ? node.typeName.right : node;
+                    errorOrSuggestion(/* isError */ false, diagLocation, Diagnostics._0_is_deprecated, symbol.escapedName as string);
+                }
+                if (type.flags & TypeFlags.Enum && symbol.flags & SymbolFlags.EnumMember) {

```

Comment:
Might as well make the type checker happy here instead of squelching it:

```suggestion
                const symbol = getNodeLinks(node).resolvedSymbol;
                if (symbol?.isDeprecated) {
                    const diagLocation = isTypeReferenceNode(node) && isQualifiedName(node.typeName) ? node.typeName.right : node;
                    errorOrSuggestion(/* isError */ false, diagLocation, Diagnostics._0_is_deprecated, symbol.escapedName as string);
                }
                if (type.flags & TypeFlags.Enum && symbol && symbol.flags & SymbolFlags.EnumMember) {
```

(`symbol?.flags` also works but might result in `undefined & 4`, which is not great)